CodeEdit/NavigatorSidebar/SourceControlNavigator/Model/SourceControlModel.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  SourceControlModel.swift
+//  CodeEdit
+//
+//  Created by Nanashi Li on 2022/05/05.
+//
+
+import Foundation
+import Git
+
+/// This model handle the fetching and adding of changes etc... for the
+/// Source Control Navigator
+public final class SourceControlModel: ObservableObject {
+
+    /// A GitClient instance
+    let gitClient: GitClient
+
+    /// The base URL of the workspace
+    let workspaceURL: URL
+
+    /// A list of changed files
+    @Published
+    public var changed: [ChangedFiles]
+
+    /// Initialize with a GitClient
+    /// - Parameter workspaceURL: the current workspace URL
+    ///
+    public init(workspaceURL: URL) {
```

Comment:
is the workspaceURL here used only for initializing the GitClient ? If so we should inject the git client instead of the workspaceURL

Reply:
It's used for `ChangedFileItemView` to open files in the finder also

Reply:
marking as unresolved since it wasn't resolved. The `ChangedFileItemView` seems to already have a workspaceURL